src/hookEvent.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef DISPLAZ_HOOKEVENT_H_INCLUDED
+#define DISPLAZ_HOOKEVENT_H_INCLUDED
+
+#include <QByteArray>
+#include <QKeySequence>
+#include <QShortcut>
+
+/// Class for registering custom hook events. Implemented via QShortCut,
+/// i.e. for QKeySequences only.
+class hookEvent : public QObject
+{
+    Q_OBJECT
+    public:
+        hookEvent(QObject* parent = NULL);
+
+        void setHookEvent(QByteArray hookSpec);
+        int getHookId() const { return latestHookIndex; }
+        QVector<QShortcut*> m_shortCut;
```

Comment:
Right, forgot to make it private and add an access member function after I was done experimenting.